server/src/main/java/org/apache/iotdb/db/mpp/execution/operator/process/last/UpdateLastCacheOperator.java

```diff
@@ -106,7 +107,14 @@ public TsBlock next() {
 
     if (needUpdateCache) {
       TimeValuePair timeValuePair = new TimeValuePair(lastTime, lastValue);
-      lastCache.updateLastCache(fullPath, timeValuePair, false, Long.MIN_VALUE);
+      lastCache.updateLastCache(
+          ((DataDriverContext) operatorContext.getInstanceContext().getDriverContext())
+              .getDataRegion()
+              .getStorageGroupName(),
```

Comment:
since sg name won't change, you should make it as a field of operator instance instead of getting it through function call each time.

Reply:
fixed